KUBERNETES.md

```diff
@@ -0,0 +1,134 @@
+# Modifying Kubernetes objects
+
+You can delegate Kubernetes object manipulation to the shell-operator.
+
+To do this, you have to write one or more JSON documents describing operation type and its parameters to a file.
+List of possible operations and corresponding JSON specification can be found below.
+
+The path to the file is found in the `$KUBERNETES_OPERATIONS_PATH` environment variable.
+
+> TODO: bash examples
+
+## Operations
+
+### Create
+
+* `operationType` — specifies an operation's type. Deletion types map directly to Kubernetes
+  DELETE's [`propagationPolicy`](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/).
+    * `CreateOrUpdate` — accept a Kubernetes object.
+      It retrieves an object, and if it already exists, computes a JSON Merge Patch and applies it (will not update .status field).
+      If it does not exist, we create the object.
+    * `Create` — will fail if an object already exists
```

Comment:
Does this fail will lead to a hook error, so the hook will be restarted?

Reply:
It will lead to the hook error.